core/reactor_common.c

```diff
@@ -988,75 +989,59 @@ trigger_handle_t _lf_schedule(environment_t *env, trigger_t* trigger, interval_t
-                    // existing->time < env->current_tag.time case below).
-                    // NOTE: Because microsteps are not explicit, if the tag of
-                    // the preceding event is equal to the current time, then
-                    // we search the event queue to figure out whether it has
-                    // been handled yet.
-                    if (existing->time > env->current_tag.time ||
-                            (existing->time == env->current_tag.time &&
-                            pqueue_find_equal_same_priority(env->event_q, existing) != NULL)) {
+                    // If the event with the previous time is still on the event
+                    // queue, then replace the token.  To find this event, we have
+                    // to construct a dummy event_t struct.
+                    event_t* dummy = _lf_get_new_event(env);
+                    dummy->next = NULL;
+                    dummy->trigger = trigger;
+                    dummy->time = trigger->last_time;
+                    event_t* found = (event_t *)pqueue_find_equal_same_priority(env->event_q, dummy);
+
+                    if (found != NULL) {
                         // Recycle the existing token and the new event
                         // and update the token of the existing event.
-                        _lf_replace_token(existing, token);
+                        _lf_replace_token(found, token);
                         _lf_recycle_event(env, e);
+                        _lf_recycle_event(env, dummy);
+                        // Leave the last_time the same.
                         return(0);
                     }
                     // If the preceding event _has_ been handled, then adjust
                     // the tag to defer the event.
                     intended_time = earliest_time;
                     break;
                 default:
-                    if (existing->time == env->current_tag.time &&
-                            pqueue_find_equal_same_priority(env->event_q, existing) != NULL) {
-                        if (_lf_is_tag_after_stop_tag(env, (tag_t){.time=existing->time,.microstep=env->current_tag.microstep+1})) {
-                            // Scheduling e will incur a microstep at timeout, 
-                            // which is illegal.
-                            _lf_recycle_event(env, e);
-                            return 0;
-                        }
-                        // If the last event hasn't been handled yet, insert
-                        // the new event right behind.
-                        existing->next = e;
-                        return 0; // FIXME: return a value
-                    } else {
-                         // Adjust the tag.
-                        intended_time = earliest_time;
-                    }
+                    // Default policy is defer
+                    intended_time = earliest_time;
```

Comment:
We had decided that "replace" should become the default (see https://github.com/lf-lang/lingua-franca/issues/1464#issuecomment-1797990069). I think it would make that change as part of PR and I can help with that.

Reply:
Actually, "replace" has a very weird feature, one that convinces me it should **not** be the default.  In particular, I can ask for a value to be scheduled at, say, (1000, 0), and because of the "replace" policy, the value will appear at (500, 0). Time travel!

In any case, that change, if we make it, should be a separate PR and should be coupled with the other changes in [1464](https://github.com/lf-lang/lingua-franca/issues/1464#issuecomment-1797990069), I think.  I don't think we should slow down this PR for it.

Reply:
The reason I'm making this connection is that we have several very long threads discussing weird corner cases and making proposals for changes. The problem with these threads is that they tend to linger, and, over time, related issues are then discussed independently, completely ignoring progress that previously had been made...

I'm not proposing a hold up, and wanted to get right to it, but your comment just now gives me pause... Have you gone over the thread that I linked to?

Reply:
> Actually, "replace" has a very weird feature, one that convinces me it should **not** be the default. In particular, I can ask for a value to be scheduled at, say, (1000, 0), and because of the "replace" policy, the value will appear at (500, 0). Time travel!

I'm not sure that the behavior you're describing is implied by the term "replace." What if we don't only update the value, but also the tag (i.e., a wholesale replacement of the event)?

Also, it seems that most weird scenarios are tied with non-monotonic event sequences to begin with, which seem atypical.

Reply:
The "replace" in [1464](https://github.com/lf-lang/lingua-franca/issues/1464#issuecomment-1797990069) is not referring the policy to use to handle `min_spacing`.  That issue is talking about something else altogether, which is what to do when two `lf_schedule` calls on the same action are requesting an event at the same future tag.  The proposal in [1464](https://github.com/lf-lang/lingua-franca/issues/1464#issuecomment-1797990069) is that the second event will replace the first.  This would be largely independent of the `min_spacing` property and its policy.

The `min_spacing` policy kicks in not when two calls to `lf_schedule` ask for the same tag, but when they as for two _times_ that are too close together.  The `replace` policy could do one of two things:

1. Replace the earlier event's value with the new value (what this PR does), or
2. Remove the earlier event and insert the new event at the later tag.

Both of these have undesirable features.  The first does time travel, the second discards an earlier event and therefore slows down responses.  In fact, the second one could result in a system _never_ responding because each event gets replaced with a later event, and before the later event can be replaced, it too gets replaced.  This is nastier than time travel, IMHO, so I opted for 1.

A better solution might be to eliminate the "replace" policy altogether.  That would be better than making it the default.

Reply:
I think that there is merit in both of the replace policy actions that are mentioned. The developer may want one behavior over the other depending on the exact requirement and what tradeoff is acceptable.

As such would it be appropriate to suggest that the _current_ `replace` policy gets renamed to an `update` policy which may convey the actual action of what is happening a bit better. 
And a new `replace` policy is added that implements the 2. point of replacing the previous event in its entirety thus acting at the later time.

If the `replace` policy is eliminated then I believe it is necessary for the ability to control event queues, such as removing an event. To be able to remove stale data from a queue for newer data to be used as an example. 
But should that default policy change even be for all actions or an action with the same time but different tag? As my reading of the proposal looks like it morphed from scheduling actions at the same time and how to handle that to the default policy for all actions.

Reply:
I agree that update would be better name for the replace policy.  But I'm not sure we want to even offer option 2.  The risk of having a never-ending chain of never processed events seems like it could create safety risks.

Reply:
This PR seems to be stalled on what I consider to be a secondary issue that should be split off into a separate issue.  At most, this PR could do the rename of "replace" policy to "update" policy.  However, such a change is not backward compatible, requires updates in two repos, and requires updating the handbook web pages.  I don't have the bandwidth to make this change right now.  Should we merge this and open an issue for that change?

Reply:
Apologies, I was not intending to suggest that the naming be dealt with currently as I am aware it is a breaking change. My biggest concern is the initial bug as it is affects some of my projects quite a bit.

Reply:
OK, I'm going to mark this resolved and move forward.